Training4/src/task-1/src/App.tsx

```diff
@@ -0,0 +1,58 @@
+import React, {useEffect, useReducer, useState, useCallback} from 'react';
+import './App.css';
+import {Joke} from "./interfaces/joke";
+import axios, {AxiosResponse} from "axios";
+import JokeCard from "./component/joke-card";
+import {jokeReducer} from "./reducer/joke-reducer";
+import throttle from "lodash.throttle";
+const base_url = 'https://official-joke-api.appspot.com/random_ten'
+
+type StateType = {
+    data?: Joke[],
+    isLoading?: boolean,
+    error?: any
+}
+
+const initState: StateType = {
+    data: [],
+    error: null
+}
+
+function App() {
+    const [isLoading, setLoading] = useState<boolean>(true)
+    const [jokes, dispatch] = useReducer(jokeReducer, initState)
+    useEffect(() => {
+        throttleApiCall()
+    }, [])
+
+    const fetchJokes = () => {
+        setLoading(true);
+        axios.get(base_url)
+            .then((res: AxiosResponse<Joke[]>) => {
+                setLoading(false);
+                dispatch({type: 'SUCCESS', data: res.data})
+            })
+            .catch((err) => {
+                dispatch({type: 'ERROR', data: err})
+            })
+    }
+    const throttleApiCall = useCallback(throttle(fetchJokes, 5000),[])
```

Comment:
fetchJokes is created new every render time, so you don't put it as dependencies of useCallback, which means the throttle function always applies for the first func?

Reply:
btw, throttle or debounce function should not apply for the function can be created many times